numpy/core/tests/test_ufunc.py

```diff
@@ -866,6 +927,89 @@ def test_innerwt_empty(self):
         w = np.array([], dtype='f8')
         assert_array_equal(umt.innerwt(a, b, w), np.sum(a*b*w, axis=-1))
 
+    def test_cross1d(self):
+        """Test with fixed-sized signature."""
+        a = np.eye(3)
+        assert_array_equal(umt.cross1d(a, a), np.zeros((3, 3)))
+        out = np.zeros((3, 3))
+        result = umt.cross1d(a[0], a, out)
+        assert_(result is out)
+        assert_array_equal(result, np.vstack((np.zeros(3), a[2], -a[1])))
+        assert_raises(ValueError, umt.cross1d, np.eye(4), np.eye(4))
+        assert_raises(ValueError, umt.cross1d, a, np.arange(4.))
+        assert_raises(ValueError, umt.cross1d, a, np.arange(3.), np.zeros((3, 4)))
+
+    def test_can_ignore_signature(self):
```

Comment:
Similar thought to above re: potential advantage of splitting into at least a few more specialized smaller tests and / or using some parametrization.

Reply:
This one is more difficult since each one does something slightly different and there is a natural progression from the simple to the complex, mangling the input differently at each step. Can we leave it for later?

Reply:
I also think this would be clearer as separate tests. You'd end up with a handful more array initialization, but it would make it much clearer what is being tested.